log-appender/src/main/scala/io/phdata/pulse/Stream.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018 phData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.phdata.pulse
+
+import java.net.URI
+import java.util.concurrent.TimeUnit
+
+import io.phdata.pulse.log.{ HttpManager, JsonParser }
+import monix.reactive.subjects.ConcurrentSubject
+import monix.execution.Scheduler.Implicits.global
+import org.apache.http.impl.client.{ CloseableHttpClient, HttpClientBuilder }
+import org.apache.log4j.spi.LoggingEvent
+
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+
+trait Stream[E] {
+  var batchFlushDuration: FiniteDuration = Duration(1, TimeUnit.SECONDS)
+  var batchFlushSize: Int                = 100
+
+  // Create a subject we can write events to
+  val subject = ConcurrentSubject.publish[E]
+
+  subject
+    .bufferTimedAndCounted(batchFlushDuration, batchFlushSize)
+    .foreach(y => save(y))
+
+  def append(value: E): Unit = subject.onNext(value)
+
+  def save(values: Seq[E])
+}
+
+class HttpStream(flushDuration: Int, flushSize: Int, postUrl: String) extends Stream[LoggingEvent] {
+  this.batchFlushDuration = Duration(flushDuration, TimeUnit.SECONDS)
```

Comment:
Is it possible to create the Duration in the Java code and pass it as an argument? That would make the intent of the argument clear and it would be impossible to accidentaly swap flushDuration and flushSize. They are both ints so they could get swapped.